src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6740,8 +6825,76 @@ private static bool ReportRefOrOutArgument(AnalyzedArguments analyzedArguments,
             return false;
         }
 
+        /// <summary>
+        /// PROTOTYPE: remove this method before shipping (this is for demo purposes only)
+        /// </summary>
+        private BoundExpression GetPrototypeExtensionIndexerOpt(ExpressionSyntax syntax, BoundExpression receiver, AnalyzedArguments analyzedArguments, DiagnosticBag diagnostics)
+        {
+            // can only have one unnamed argument, with no ref kind.
+            if (analyzedArguments.Arguments.Count != 1
+                || analyzedArguments.Argument(0).Type is null
+                || analyzedArguments.RefKinds.Any()
+                || analyzedArguments.Names.Any())
+            {
+                return null;
+            }
+
+            BoundExpression argument = analyzedArguments.Argument(0);
+
+            // argument must be a range or an index
+            if (argument.Type != GetWellKnownType(WellKnownType.System_Index, diagnostics, syntax)
```

Comment:
>diagnostics [](start = 78, length = 11)

It feels unexpected that we would preserve diagnostics here. If these types are not available, will any indexing operation report an error?

Reply:
Already fixing that :) my build caught it.

---
In reply to: [185364143](https://github.com/dotnet/roslyn/pull/26405#discussion_r185364143) [](ancestors = 185364143)